tests/e2e/actions.go

```diff
@@ -268,72 +391,225 @@ type SubmitConsumerAdditionProposalAction struct {
+	}
+
+	// Send consumer chain registration
+	cmd := tr.target.ExecCommand(
+		tr.testConfig.chainConfigs[providerChain].BinaryName,
+		"tx", "provider", "create-consumer", jsonFile,
+		`--from`, `validator`+fmt.Sprint(validator),
+		`--chain-id`, string(tr.testConfig.chainConfigs[providerChain].ChainId),
+		`--home`, tr.getValidatorHome(providerChain, validator),
+		`--gas`, `900000`,
+		`--node`, tr.getValidatorNode(providerChain, validator),
+		`--keyring-backend`, `test`,
+		"--output", "json",
+		`-y`,
+	)
+
+	fmt.Println("Create consumer", "cmd", cmd.String())
+	fmt.Println("Create consumer", "json", string(content))
+
+	bz, err = cmd.CombinedOutput()
+	if err != nil {
+		log.Fatal("create consumer failed ", "error: ", err, "output: ", string(bz))
+	}
+
+	fmt.Println("create consumer output=", string(bz))
+
+	txResponse := &TxResponse{}
+	err = json.Unmarshal(bz, txResponse)
+	if err != nil {
+		log.Fatalf("unmarshalling tx response on consumer-registration: %s, json: %s", err.Error(), string(bz))
+	}
+
+	if txResponse.Code != 0 {
+		log.Fatalf("sending transaction failed with error code %d, Log:'%s'", txResponse.Code, txResponse.RawLog)
+	}
+
+	// TODO: introduce waitForTx
+	tr.waitBlocks(providerChain, 2, 10*time.Second)
+
+	// Get Consumer ID from transaction
+	cmd = tr.target.ExecCommand(
+		tr.testConfig.chainConfigs[providerChain].BinaryName,
+		"query", "tx", txResponse.TxHash,
+		`--node`, tr.getValidatorNode(providerChain, validator),
+		"--output", "json",
+	)
+	bz, err = cmd.CombinedOutput()
+	if err != nil {
+		log.Fatal("not able to query tx containing consumer registration: cmd:", cmd, "err:", err.Error(), "out:", string(bz))
+	}
+
+	err = json.Unmarshal(bz, txResponse)
+	if err != nil {
+		log.Fatalf("unmarshalling tx containing consumer registration: %s, json: %s", err.Error(), string(bz))
+	}
+
+	consumer_id := ""
+	for _, event := range txResponse.Events {
+		if event.Type != "consumer_creation" {
+			continue
+		}
+		attr, exists := event.GetAttribute("consumer_id")
+		if !exists {
+			log.Fatalf("no event with consumer_id found in tx content of consumer-registration: %v", event)
+		}
+		consumer_id = attr.Value
+	}
+	if consumer_id == "" {
+		log.Fatalf("no consumer-id found in consumer creation transaction events for chain '%s'. events: %v", consumerChain, txResponse.Events)
+	}
+
+	cfg, exists := tr.testConfig.chainConfigs[e2e.ChainID(chainID)]
+	if !exists {
+		log.Fatal("no chain config found for consumer chain", chainID)
+	}
+	if cfg.ConsumerId != "" && cfg.ConsumerId != e2e.ConsumerID(consumer_id) {
+		log.Fatal("chain ", chainID, " registered already with a different consumer ID", consumer_id)
+	}
+
+	// Set the new created consumer-id on the chain's config
+	cfg.ConsumerId = e2e.ConsumerID(consumer_id)
+	tr.testConfig.chainConfigs[e2e.ChainID(chainID)] = cfg
+
+	return consumer_id
+}
+
+// submitConsumerAdditionProposal initializes a consumer chain and submits a governance proposal
```

Comment:
This submit proposal to add a consumer chain, and hence the name `submitConsumerAdditionProposal` is technically correct. Nevertheless, I was wondering if we should rename all those functions (e.g., `submitConsumerRemovalProposal`, etc.) and all those actions (e.g., `SubmitConsumerRemovalProposalAction`, etc.) because naming-wise they still refer to the old non-permissionless way.

Reply:
don't want to do that (at least for now) because of compatibility tests where Action/Steps are shared accross versions
At least that's what I had in mind for the upcoming fixes on the compatibility tests.

Reply:
Is this something we could do in the future? Then, can we create an issue for this? We don't have to do this here.
But currently, naming is inconsistent, e.g., we have `ConsumerUpdateProposal` and `ConsumerModificationProposal`, etc.

Reply:
opened an issue: https://github.com/cosmos/interchain-security/issues/2198